test/helper/zone-helper.spec.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+describe('zone helper util test', () => {
+  const getNonZoneAwareStack: (err: Error) => string = Error['getNonZoneAwareStack'];
+  const zoneAwareFrames = [
+    'Zone.run', 'Zone.runGuarded', 'Zone.scheduleEventTask', 'Zone.scheduleMicroTask',
+    'Zone.scheduleMacroTask', 'Zone.runTask', 'ZoneDelegate.scheduleTask',
+    'ZoneDelegate.invokeTask', 'ZoneTask.invoke', 'zoneAwareAddListener', 'drainMicroTaskQueue'
+  ];
+
+  const expectStack = function(err: Error) {
+    const simpleStack = getNonZoneAwareStack(err);
+    if (!simpleStack) {
+      return;
+    }
+    const frames = simpleStack.split('\n');
+    for (let i = 0; i < frames.length; i++) {
+      expect(zoneAwareFrames.filter(f => frames[i].indexOf(f) !== -1).length).toBe(0);
+    }
+  };
+
+  const errorZoneSpec = {
+    name: 'errorZone',
+    done: null,
+    onHandleError: (parentDelegate, currentZone, targetZone, error) => {
+      expectStack(error);
+      setTimeout(() => {
+        errorZoneSpec.done && errorZoneSpec.done();
+      }, 0);
+      return false;
+    }
+  };
+
+  const errorZone = Zone.root.fork(errorZoneSpec);
+
+  const errorTest = function(testFn: Function) {
+    return function(done) {
+      errorZoneSpec.done = done;
+      errorZone.run(testFn);
+    };
+  };
+
+  it('test getNonZoneAwareStack', errorTest(() => {
+       setTimeout(() => {
+         throw new Error('test error');
+       }, 10);
+     }));
+
+  it('test promise getNonZoneAwareStack', (done) => {
+    const p = new Promise((resolve, reject) => {
+      reject(new Error('test error'));
+    });
+    p.catch(err => {
+      expectStack(err);
+      done();
+    });
+  });
+
+  it('test eventTask getNonZoneAwareStack', errorTest(() => {
+       const task = Zone.current.scheduleEventTask('errorEvent', () => {
+         throw new Error('test error');
+       }, null, () => null, null);
+       task.invoke();
+     }));
+
+  it('test longStackTrace getNonZoneAwareStack', errorTest(() => {
+       const task =
+           Zone.current.fork(Zone['longStackTraceZoneSpec']).scheduleEventTask('errorEvent', () => {
```

Comment:
The stack frames will be a little different when error occurs in zoneSpec has callback and without callback. So I write two cases for it. I will add comment to it.